chain/chain/src/resharding/trie_state_resharder.rs

```diff
@@ -278,3 +289,194 @@ impl TrieStateResharderMetrics {
+
+    use crate::runtime::NightshadeRuntime;
+    use crate::types::ChainConfig;
+
+    use super::*;
+
+    type KeyValues = Vec<(Vec<u8>, Option<Vec<u8>>)>;
+    struct TestSetup {
+        runtime: Arc<dyn RuntimeAdapter>,
+        initial: KeyValues,
+        parent_shard: ShardUId,
+        left_shard: ShardUId,
+        right_shard: ShardUId,
+        left_root: CryptoHash,
+        right_root: CryptoHash,
+    }
+
+    impl TestSetup {
+        fn as_status(&self) -> TrieStateReshardingStatus {
+            TrieStateReshardingStatus::new(
+                self.parent_shard,
+                TrieStateReshardingChildStatus::new(self.left_shard, self.left_root),
+                TrieStateReshardingChildStatus::new(self.right_shard, self.right_root),
+            )
+        }
+    }
+
+    fn setup_test() -> TestSetup {
+        let shard_layout = ShardLayout::v2(vec![], vec![ShardId::new(0)], None);
+        let genesis = Genesis::from_accounts(
+            Clock::real(),
+            vec!["aa".parse().unwrap()],
+            1,
+            shard_layout.clone(),
+        );
+        let tempdir = tempfile::tempdir().unwrap();
+        let store = create_test_store();
+
+        // This enables the flat storage and in-memory tries for testing.
+        // It sets FlatStorageReadyStatus and creates an empty ChunkExtra.
+        let _tries = TestTriesBuilder::new()
+            .with_shard_layout(shard_layout.clone())
+            .with_store(store.clone())
+            .with_flat_storage(true)
+            .with_in_memory_tries(true)
+            .build();
+
+        let epoch_manager = EpochManager::new_arc_handle(store.clone(), &genesis.config, None);
+        let runtime =
+            NightshadeRuntime::test(tempdir.path(), store, &genesis.config, epoch_manager);
+
+        let parent_shard = ShardUId { version: 3, shard_id: 0 };
+        let tries = runtime.get_tries();
+        tries.load_memtrie(&parent_shard, None, false).unwrap();
+
+        let make_account_key = |i: usize| {
+            let account_str = format!("account-{:02}", i);
+            TrieKey::Account { account_id: account_str.parse().unwrap() }
+        };
+
+        // Using `TrieKey::Account` to create the trie keys. This is so in
+        // resharding, the keys can be split by `AccountId`. Deliberately
+        // contains duplicate values to test reference counting
+        let initial =
+            (0..1000).map(|i| (make_account_key(i).to_vec(), Some(vec![i as u8]))).collect_vec();
+        let initial = simplify_changes(&initial); // Sorts & deduplicates the changes
+
+        let parent_root =
+            test_populate_trie(&tries, &Trie::EMPTY_ROOT, parent_shard, initial.clone());
+
+        // Create tries to represent the post-state root at the end of the resharding block.
+        // Each child shard will have its post-state.
+        let boundary_account =
+            make_account_key(initial.len() / 2 as usize).get_account_id().unwrap();
```

Comment:
If the intention was to split the shard equally, I am not sure it works, if there is (1000/2)%100=0

Reply:
it doesn't need to be completely equal, I just wanted something roughly split in half.